src/Resources/config/client.xml

```diff
@@ -6,11 +6,21 @@
 
     <services>
         <service id="Nexy\Slack\Client" public="true">
+            <argument type="service" id="nexy_slack.http.client" />
+            <argument type="service" id="nexy_slack.http.request_factory" />
+            <argument type="service" id="nexy_slack.http.stream_factory" />
             <argument>%nexy_slack.endpoint%</argument>
             <argument>%nexy_slack.config%</argument>
-            <argument type="service" id="nexy_slack.http.client" />
         </service>
 
         <service id="nexy_slack.client" alias="Nexy\Slack\Client"/>
+
+        <service id="nexy_slack.request_factory.default" class="Psr\Http\Message\RequestFactoryInterface">
```

Comment:
HttplugBundle doesn't provide services directly creating the PSR-17 factories, so the services here are the defaults used to discover compatible factories through the Httplug discovery system.